src/app/components/client/ExposureCard.tsx

```diff
@@ -45,15 +45,44 @@ export type ExposureCardProps = {
 };
 
 export const ExposureCard = ({ exposureData, ...props }: ExposureCardProps) => {
-  return isScanResult(exposureData) ? (
-    <ScanResultCard {...props} scanResult={exposureData} />
-  ) : (
-    <SubscriberBreachCard {...props} subscriberBreach={exposureData} />
-  );
+  const [dataBrokerImage, setDataBrokerImage] = useState<ReactNode>();
+
+  useEffect(() => {
+    const loadDataBrokerImage = async () => {
+      if (isScanResult(exposureData)) {
+        const cleanedName = exposureData.data_broker.split(".")[0];
+        try {
+          const logo = await import(
+            `../client/assets/data-brokers/${cleanedName}.jpg`
+          );
+          setDataBrokerImage(<Image src={logo.default} alt="" />);
```

Comment:
While I assume this works, React has [built-in affordances](https://react.dev/reference/react/lazy) to support lazily loading components, and using that presumably allows it to optimise the render order and be more robust to future updates.

I believe you would implement that something like this:

```tsx
export const DataBrokerImage = (props: { name: string }) => {
  const DataBrokerLogo = lazy(() => import(`../client/asserts/data-brokers/${name}.jpg`).then(logo => ({
    default: <Image src={logo.default} alt="" />,
  }));

  return (
    <Suspense fallback={<FallbackLogo name={name}/>}>
      <DataBrokerLogo/>
    </Suspense>
  );
};
```

And then you should be able to pass `<DataBrokerImage name={cleanedName} />` into `ScanResultCard`'s `exposureImg`.

Reply:
Yeah that's right, that's why in my example code the function I passed into `lazy` didn't just the `import` of the image, but then also wrapped it in a component and returned _that_ on the `default` property. Maybe it's easier to follow if I split it up:

```ts
async function importImageInComponent(name: string) => {
  const logo = await import(`../client/asserts/data-brokers/${name}.jpg`);
  return {
    default: <Image src={logo.default} alt="" />,
  };
};

const DataBrokerLogo = lazy(() => importImageInComponent(name));
```

And yeah, the `lazy` you're seeing in the DOM is a different one: [this one](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Img#lazy) rather than [this one](https://react.dev/reference/react/lazy). They do similar things in different situations, hence the similar names.